src/Raven.Server/Documents/Indexes/Static/Extensions/MetadataExtensions.cs

```diff
@@ -1,38 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-
-namespace Raven.Server.Documents.Indexes.Static.Extensions
-{
-    public static class MetadataExtensions
-    {
-        public static IEnumerable<dynamic> WhereEntityIs(this IEnumerable<dynamic> self, params string[] collections)
```

Comment:
to match this?

Reply:
I think that those are two different things. `WhereEntityIs` is filtering the collections on which the index is working and `IfEnitytIs` is useful when doing `let x = ` to get appropriate documents in there.

Reply:
IIRC, that was to be able to do something like `docs.WhereEntityIs("Abc", "Def", etc)`

Reply:
I do not follow. What's the purpose of `IfEntityIs` then?

Reply:
```
IEnumerable<TResult> WhereEntityIs<TResult>(this IEnumerable<object> queryable, params string[] names)

TResult IfEntityIs<TResult>(this object queryable, string name)
```

notice the difference. This is why `WhereEntityIs` is good at `from doc in docs.WhereEntityIs()` and the other one IMO for
```
from doc in docs.WhereEntityIs("Dogs", "Cats")
let dog = doc.IfEntityIs("Dogs")
let cat = doc.IfEntityIs("Cats")
select ...
```

Reply:
And we can add `params` to `IfEntityIs` although I'm not sure if this is that usable.

Reply:
OK